### DISCUSSION TITLE
Exclude files from Storage::files()

### DISCUSSION THREAD

**Discussion by LeMatosDeFuk** [laravel/framework#48550] [Ideas] [answered]
Hi,

I recently discovered that the Storage::files() method doesn't provide an option to exclude specific files from its results. For instance, if there's a .gitignore file in the directory, it appears there's currently no direct way to filter it out.

What if we introduced an except method to enhance the files() method? The usage could be as straightforward as:

`Storage::files($dir)->except(['.gitignore', 'otherFile.ext']);
`

**Comment by henzeb** [comment]
There is a lot that's missing in Storage, like glob. Which is in the File Facade. For local operations you can use that facade, but for external operations this is a problem...

**Comment by shaedrich** [comment, marked as answer]
I suppose, that's not implemented in [League's](https://thephpleague.com) [Flysystem](https://flysystem.thephpleague.com/docs/usage/filesystem-api/). Maybe you can [file a feature request over there](https://github.com/thephpleague/flysystem/issues).
